clippy_lints/src/needless_bool.rs

```diff
@@ -226,6 +226,11 @@ fn check_comparison<'a, 'tcx>(
 
     if let ExprKind::Binary(_, ref left_side, ref right_side) = e.node {
         let (l_ty, r_ty) = (cx.tables.expr_ty(left_side), cx.tables.expr_ty(right_side));
+
+        if in_macro(left_side.span) || in_macro(right_side.span) {
```

Comment:
This is a little bit to restrictive: [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=b24aa44c211d85a04f7283f11f6395f3)

This won't suggest anything for `macro! == true`. But this could be simplified to `macro!`.

---
While testing this, I found another "bug". Clippy suggests for this:
```
warning: equality checks against true are unnecessary
 --> src/main.rs:8:8
  |
8 |     if m!(func) == true {
  |        ^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `$func()`
  |
  = note: #[warn(clippy::bool_comparison)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
```

This suggestion is not applicable, obviously. This can be fixed by changing this line
https://github.com/rust-lang/rust-clippy/blob/1cdac4a9c798f8e22a4db220020713175eb9159b/clippy_lints/src/needless_bool.rs#L271
with 
https://github.com/rust-lang/rust-clippy/blob/3e760cc93a4820c1f8fbb0d5e125567e065776d3/clippy_lints/src/utils/sugg.rs#L84

But that won't be that easy, since we need a function, which adapts the applicability. So either
1. you can try to fix this in this PR
2. you can open an issue and leave this for later

Reply:
Thanks for the review!

I'd love to fix this in this PR. :)